src/pymor/models/iosys.py

```diff
@@ -928,6 +930,43 @@ def get_ast_spectrum(self, ast_pole_data=None, mu=None):
 
             return ast_lev, ast_ews[idx], ast_rev
 
+    def moebius_substitution(self, M, sampling_time=0):
+        assert isinstance(M, MoebiusTransform)
+        if not isinstance(self.E, IdentityOperator):
+            return NotImplemented
+
+        a, b, c, d = M.coefficients
+        s = a * d - b * c
+        v = np.sqrt(np.abs(s))
+        Gamma = InverseOperator(a * self.E - c * self.A)
+        At = NumpyMatrixOperator(
+            M.apply_inverse(M.source.from_numpy(self.A.matrix)).to_numpy().T,
+            source_id=self.A.source_id,
+            range_id=self.A.range_id)
```

Comment:
I have added a basic implementation of `assemble_concat` in #1618. Maybe something similar could be done for the inverse of a `NumpyMatrixOperator`?

Reply:
How about moving `Gamma` to the `E` matrix by state-space transformation? It looks like instead using

```python
Et = a * self.E - c * self.A
At = a * self.A - b * self.E
Ct = v * self.C
```

defines an equivalent system.

Also, it would be good to allow for `self.E` to not necessarily be an `IdentityOperator`.

Reply:
Hmm. I'm not sure I get what you mean. I can construct a system with
```
        Et = d * self.E + c * self.A
        At = (a * self.A + b * self.E)
        Bt = np.sign(s) * v * self.B 
        Ct = v * self.C @ InverseOperator(Et)
        Dt = self.D - c * self.C @ InverseOperator(Et) @ self.B
```
But that still leaves the inverse in `Ct` and `Dt`...

Reply:
I don't fully understand what is going on but it seems like having the inverses just in `Ct` and `Dt` solves the problem with the slow transfer function evaluations. So I guess it works out like this, thanks for the suggestion!

Reply:
The speed improvement is good. It is expected because the previous implementation involved solving a linear system with a lot of right-hand sides (basically, `Et.apply_inverse(At)`).

> > Also, it would be good to allow for self.E to not necessarily be an IdentityOperator.
> 
> Yes definitely. So far, I only have that one source that does things in absence of `E`. I'm not sure if/how these results can be generalized at the moment.

I guess that (using pencil and paper) you could apply the Moebius transform to (E<sup>-1</sup> A, E<sup>-1</sup> B, C), and then see if you can get rid of E<sup>-1</sup> in the transformed system. This should then generalize the procedure to systems where E is not the identity.